src/pybind/mgr/dashboard/frontend/src/app/ceph/cephfs/cephfs-subvolume-list/cephfs-subvolume-list.component.html

```diff
@@ -74,3 +74,57 @@
             *ngIf="row.info.pool_namespace"
             [tooltipText]="row.info.pool_namespace"></cd-label>
 </ng-template>
+
+<ng-template #removeModal>
+  <cd-modal [modalRef]="activeModal">
+    <ng-container class="modal-title">Remove Subvolume {{selectedName}}</ng-container>
+
+    <ng-container class="modal-content">
+      <form name="removeForm"
+            #formDir="ngForm"
+            [formGroup]="removeForm"
+            novalidate>
+        <div class="modal-body">
+          <p>Do you want to remove subvolume <strong>{{selectedName}}</strong>?</p>
+
+          <div class="custom-control custom-checkbox">
+            <input type="checkbox"
+                   class="custom-control-input"
+                   name="retainSnapshots"
+                   id="retainSnapshots"
+                   formControlName="retainSnapshots">
+            <label class="custom-control-label"
+                   for="retainSnapshots"
+                   i18n>Retain snapshots <cd-helper>The subvolume can be removed retaining
+                     existing snapshots using this option.
+                      If snapshots are retained, the subvolume is considered empty for all
+                       operations not involving the retained snapshots.</cd-helper></label>
+          </div>
+
+          <div class="custom-control custom-checkbox">
+            <input type="checkbox"
+                   class="custom-control-input"
+                   name="confirmation"
+                   id="confirmation"
+                   formControlName="confirmation"
+                   autofocus>
+            <label class="custom-control-label"
+                   for="confirmation"
+                   i18n>Yes, I am sure.</label>
+          </div>
+        </div>
+
+        <cd-alert-panel *ngIf="errorMessage.length > 1"
+                        type="error">
+            {{errorMessage}}
+        </cd-alert-panel>
```

Comment:
Also I would prefer this to be displayed on top of the modal. which will match our other areas! And this should be inside the `modal-body`. From the video recording it seems like it is outside of it